sygnal/firebasepushkin.py

```diff
@@ -13,259 +13,419 @@
-        if body_prefix:
-            notification_body = f"{body_prefix} {notification_body}"
-        if "images" in message:
-            notification_body += self.config.message_types.get("m.image")
-        elif "videos" in message:
-            notification_body += self.config.message_types.get("m.video")
-        elif "title" not in message and "message" in message:
-            notification_body += message["message"].strip()
-
-        return self.default_notification(
-            data, notification_title, unread_count, pushkeys, notification_body
-        )
+        Args:
+            n: The incoming notification for which the data should be generated
 
-    def default_notification(
-        self, data, notification_title, unread_count, pushkeys, notification_body
-    ):
-        return messaging.MulticastMessage(
-            notification=messaging.Notification(
-                title=notification_title, body=notification_body
-            ),
-            data={
-                "title": notification_title,
-                "body": notification_body,
-                "room_id": data["room_id"],
-            },
-            android=messaging.AndroidConfig(
-                priority=data["prio"],
-                notification=messaging.AndroidNotification(
-                    click_action="FLUTTER_NOTIFICATION_CLICK", tag=data["room_id"]
-                ),
-            ),
-            apns=messaging.APNSConfig(
-                headers={"apns-priority": "10" if data["prio"] == 10 else "5"},
-                payload=messaging.APNSPayload(
-                    aps=messaging.Aps(badge=unread_count, thread_id=data["room_id"])
-                ),
-            ),
-            tokens=pushkeys,
-        )
+        Returns:
+            (dict of str: obj): Containing data payload for the outgoing notification
+        """
+        data = {}
+        for field in NOTIFICATION_DATA_INCLUDED:
+            value = getattr(n, field, None)
+            if value is not None:
+                data[field] = value
+        return data
 
+    @staticmethod
+    def _event_data_from_notification(n):
+        """
+        Generates the data payload for an outgoing event type notification
+        including minimal information about the event
 
-def decode_complex_message(message: str) -> Optional[Dict]:
-    """
-    Tries to parse a message as json
+        Args:
+            n: The incoming notification for which the data should be generated
 
-    :param message: json string of notification m.text message
-    :return: dict if successful and None if parsing fails or message is not valid
-    """
-    try:
-        decoded_message = json.loads(message)
-        if is_valid_matrix_complex_message(decoded_message):
-            return decoded_message
-    except json.JSONDecodeError:
-        pass
+        Returns:
+            (dict of str: obj): Containing data payload for the outgoing notification
```

Comment:
Interestingly my IDE (PyCharm) turns this into `dict[str,str]` :thinking: 

```suggestion
            dict[str,obj]: Containing data payload for the outgoing notification
```